### ISSUE TITLE
Any idea why the plugin is not loaded

### ISSUE THREAD

**Issue by sukantasaha** [datarank/tempest#10]
Hi Installed this plugin with this command
/usr/local/elasticsearch-1.4.1-SNAPSHOT/bin/plugin -url file:/root/tempest-allocator-1.0.0.zip -install tempest
-> Installing tempest...
Trying file:/root/tempest-allocator-1.0.0.zip...
Downloading .DONE
Installed tempest into /usr/local/elasticsearch-1.4.1-SNAPSHOT/plugins/tempest

It went well
ls -al /usr/local/elasticsearch-1.4.1-SNAPSHOT/plugins/tempest/
drwxr-xr-x 10 root root  4096 Jul 21 15:14 ..
-rw-r--r--  1 root root 33341 Jul 21 15:14 tempest-allocator-1.0.0-SNAPSHOT.jar

After restarting the elasticsearch process the plugins is not loading , following from logs

loaded [clustering-carrot2, es-reporting-plugin, lang-mvel, analysis-smartcn, analysis-kuromoji, cloud-aws, analysis-icu], sites [carrot2]

Any idea why?

We are using elasticsearch-1.4.1-SNAPSHOT version

**Comment by sukantasaha** [comment]
guys can some one please help

**Comment by aewhite** [comment]
Answering for @sawyercade until he can dig deeper...

Version 1.4.1-SNAPSHOT is an old development release of ES. We have tested it out on 1.4.3 and 1.5.2. Perhaps you could try it out on a newer version? That said I have seen ES "lie" about what plugins it has loaded in past versions so it may very well be live.

**Comment by sawyercade** [comment]
I just tested on [this v.1.4.1 release], tempest loaded and executed fine.  However, @sukantasaha is right, for some reason tempest is not showing up in the list of loaded plugins. @sukantasaha , can you see if you have any lines like the following in your elasticsearch.log file? These indicate that tempest is loaded and running properly.

[2015-07-22 11:15:54,126][INFO ][com.datarank.tempest.allocator.LocalMinimumShardsAllocator] [Hugh Jones] Rebalancing.

Basically anything with com.datarank.tempest.allocator.LocalMinimumShardsAllocator shows that tempest is in the classpath and is being called properly.

I've added issue #11 for tempest not showing up in the list of loaded plugins.

**Comment by sukantasaha** [comment]
cool, thanks , I will check this and let you know

**Comment by sukantasaha** [comment]
Unfortunately its not showing,Here I paste the entire logs

[2015-07-23 13:43:13,624][INFO ][node                     ] [es-data-nqa-listening-es-cluster-1-1b] version[1.4.1], pid[20409], build[54c6d49/2015-01-06T17:52:42Z]
[2015-07-23 13:43:13,625][INFO ][node                     ] [es-data-nqa-listening-es-cluster-1-1b] initializing ...
[2015-07-23 13:43:13,706][INFO ][plugins                  ] [es-data-nqa-listening-es-cluster-1-1b] loaded [clustering-carrot2, es-reporting-plugin, lang-mvel, analysis-smartcn, analysis-kuromoji, cloud-aws, analysis-icu], sites [carrot2]
[2015-07-23 13:43:16,084][INFO ][node                     ] [es-data-nqa-listening-es-cluster-1-1b] initialized
[2015-07-23 13:43:16,084][INFO ][node                     ] [es-data-nqa-listening-es-cluster-1-1b] starting ...
[2015-07-23 13:43:16,086][INFO ][plugin.carrot2           ] [es-data-nqa-listening-es-cluster-1-1b] Resources dir: /usr/local/elasticsearch-1.4.1-SNAPSHOT/config/.
[2015-07-23 13:43:16,890][INFO ][plugin.carrot2           ] [es-data-nqa-listening-es-cluster-1-1b] Available clustering components: lingo3g, lingo, stc, kmeans, byurl
[2015-07-23 13:43:16,967][INFO ][transport                ] [es-data-nqa-listening-es-cluster-1-1b] bound_address {inet[/0:0:0:0:0:0:0:0:9300]}, publish_address {inet[/10.0.0.193:9300]}
[2015-07-23 13:43:16,976][INFO ][discovery                ] [es-data-nqa-listening-es-cluster-1-1b] nqa-listening-es-cluster/vG6ZBwrmTWC3c6jPpqmqVg
[2015-07-23 13:43:46,976][WARN ][discovery                ] [es-data-nqa-listening-es-cluster-1-1b] waited for 30s and no initial state was set by the discovery
[2015-07-23 13:43:46,985][INFO ][http                     ] [es-data-nqa-listening-es-cluster-1-1b] bound_address {inet[/0:0:0:0:0:0:0:0:9200]}, publish_address {inet[/10.0.0.193:9200]}
[2015-07-23 13:43:46,986][INFO ][node                     ] [es-data-nqa-listening-es-cluster-1-1b] started
[2015-07-23 13:44:30,771][DEBUG][action.admin.cluster.health] [es-data-nqa-listening-es-cluster-1-1b] no known master node, scheduling a retry
[2015-07-23 13:44:39,144][DEBUG][action.admin.indices.create] [es-data-nqa-listening-es-cluster-1-1b] no known master node, scheduling a retry
[2015-07-23 13:45:00,772][DEBUG][action.admin.cluster.health] [es-data-nqa-listening-es-cluster-1-1b] observer: timeout notification from cluster service. timeout setting [30s], time since start [30s]
[2015-07-23 13:45:00,802][DEBUG][action.admin.cluster.state] [es-data-nqa-listening-es-cluster-1-1b] no known master node, scheduling a retry
[2015-07-23 13:45:30,803][DEBUG][action.admin.cluster.state] [es-data-nqa-listening-es-cluster-1-1b] observer: timeout notification from cluster service. timeout setting [30s], time since start [30s]
[2015-07-23 13:45:30,854][DEBUG][action.admin.cluster.health] [es-data-nqa-listening-es-cluster-1-1b] no known master node, scheduling a retry
[2015-07-23 13:45:36,783][DEBUG][action.admin.indices.create] [es-data-nqa-listening-es-cluster-1-1b] no known master node, scheduling a retry
[2015-07-23 13:45:38,850][DEBUG][action.admin.indices.create] [es-data-nqa-listening-es-cluster-1-1b] no known master node, scheduling a retry
[2015-07-23 13:45:39,145][DEBUG][action.admin.indices.create] [es-data-nqa-listening-es-cluster-1-1b] observer: timeout notification from cluster service. timeout setting [1m], time since start [1m]
[2015-07-23 13:45:48,258][INFO ][cluster.service          ] [es-data-nqa-listening-es-cluster-1-1b] detected_master [es-master-nqa-listening-es-cluster-2-1d][1kSHbBSFSFKeilaO2j7gvw][ip-10-0-4-210][inet[/10.0.4.210:9300]]{data=false, zone=zone2, master=true}, added {[es-data-nqa-listening-es-cluster-2-1d][YD6ZxDOKT0WkmHkCPcuKeA][ip-10-0-4-97][inet[/10.0.4.97:9300]]{tag=spr-hot-tag, zone=zone2, master=false},[es-data-nqa-listening-es-cluster-9-1b][tYGYaXTERdKshz34RFL8aQ][ip-10-0-0-23][inet[/10.0.0.23:9300]]{tag=spr-cold-tag, zone=zone1, master=false},[es-master-nqa-listening-es-cluster-2-1d][1kSHbBSFSFKeilaO2j7gvw][ip-10-0-4-210][inet[/10.0.4.210:9300]]{data=false, zone=zone2, master=true},[es-data-nqa-listening-es-cluster-7-1d][jwuk5mbES3i-cwkkoyi7AA][ip-10-0-4-20][inet[/10.0.4.20:9300]]{tag=spr-cold-tag, zone=zone2, master=false},[es-data-nqa-listening-es-cluster-8-1d][BFtwHpQkSkOE5qVJ17ZHEQ][ip-10-0-4-196][inet[/10.0.4.196:9300]]{tag=spr-cold-tag, zone=zone2, master=false},[es-data-nqa-listening-es-cluster-10-1d][p9YUSUYxTniWSmLlC26-sQ][ip-10-0-4-10][inet[/10.0.4.10:9300]]{tag=spr-cold-tag, zone=zone2, master=false},[es-data-nqa-listening-es-cluster-6-1d][LppPMzBpRGOY6x8TYhQ6bg][ip-10-0-0-96][inet[/10.0.0.96:9300]]{tag=spr-hot-tag, zone=zone2, master=false},[es-master-nqa-listening-es-cluster-3-1c][OiNGvhIYTE-I3pvmnMQ6rw][ip-10-0-2-19][inet[/10.0.2.19:9300]]{data=false, zone=zone3, master=true},[es-data-nqa-listening-es-cluster-11-1b][M6VNZKqJRZCIZZisNjvAwA][ip-10-0-0-136][inet[/10.0.0.136:9300]]{tag=spr-cold-tag, zone=zone1, master=false},[es-data-nqa-listening-es-cluster-12-1b][DdFkrSQFTmSra8kSKjRbCg][ip-10-0-0-250][inet[/10.0.0.250:9300]]{tag=spr-cold-tag, zone=zone1, master=false},[es-master-nqa-listening-es-cluster-1-1b][J7C7tLgyRRWXJIIXjpB-Ng][ip-10-0-0-87][inet[/10.0.0.87:9300]]{data=false, zone=zone1, master=true},[es-data-nqa-listening-es-cluster-5-1b][caVFDb6xTumz7VrCfNzNWQ][ip-10-0-4-155][inet[/10.0.4.155:9300]]{tag=spr-hot-tag, zone=zone1, master=false},}, reason: zen-disco-receive(from master [[es-master-nqa-listening-es-cluster-2-1d][1kSHbBSFSFKeilaO2j7gvw][ip-10-0-4-210][inet[/10.0.4.210:9300]]{data=false, zone=zone2, master=true}])
[2015-07-23 13:45:48,295][INFO ][cluster.routing.allocation.decider] [es-data-nqa-listening-es-cluster-1-1b] updating [cluster.routing.allocation.disk.threshold_enabled] from [true] to [true]
[2015-07-23 13:45:48,295][INFO ][cluster.routing.allocation.decider] [es-data-nqa-listening-es-cluster-1-1b] updating [cluster.routing.allocation.disk.watermark.low] to [85%]
[2015-07-23 13:45:48,295][INFO ][cluster.routing.allocation.decider] [es-data-nqa-listening-es-cluster-1-1b] updating [cluster.routing.allocation.disk.watermark.high] to [90%]
[2015-07-23 13:45:48,295][INFO ][indices.recovery         ] [es-data-nqa-listening-es-cluster-1-1b] updating [indices.recovery.max_bytes_per_sec] from [20mb] to [10gb]
[2015-07-23 13:45:48,296][INFO ][indices.recovery         ] [es-data-nqa-listening-es-cluster-1-1b] updating [indices.recovery.concurrent_streams] from [3] to [24]

In the yml file added these lines

cluster.routing.allocation.type: com.datarank.tempest.allocator.LocalMinimumShardsAllocator
cluster.routing.allocation.probabilistic.range_ratio : 1.5
cluster.routing.allocation.probabilistic.iterations : 3000

**Comment by sawyercade** [comment]
Hmm that is problematic indeed! Can you provide your cluster setup? The big thing is that tempest needs to be installed on all nodes where node.data = true. The logs you posted appear to be from the node "es-data-nqa-listening-es-cluster-1-1b", but the master node is "es-master-nqa-listening-es-cluster-2-1d." Can you also verify that tempest was installed on any node that might be elected master, and that it's not showing up in the logs for the currently-elected master?

**Comment by aewhite** [comment]
I'll add furthermore that the balancer logs will only be emitted from the active master. I would use the head plugin or logs to verify the elected master and look there.

**Comment by sawyercade** [comment]
@sukantasaha Did you have a chance to look at the log output from your master node?

**Comment by aewhite** [comment]
This is fixed in Tempest 2.x
